File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/InvalidStateException.java
package com.github.dedis.popstellar.utility.error;

import com.github.dedis.popstellar.model.network.method.message.data.Data;

/** This exception is raised when a message is handled at the wrong state of an object */
public class InvalidStateException extends DataHandlingException {

  public InvalidStateException(Data data, String object, String state, String expected) {
    super(
        data,
        "The " + object + " is in the wrong state. It is in " + state + " expected " + expected);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/InvalidStateException.kt
package com.github.dedis.popstellar.utility.error

import com.github.dedis.popstellar.model.network.method.message.data.Data

/** This exception is raised when a message is handled at the wrong state of an object  */
class InvalidStateException(data: Data, `object`: String, state: String, expected: String) :
    DataHandlingException(
        data,
        "The $`object` is in the wrong state. It is in $state expected $expected"
    )